misc.py

```diff
@@ -0,0 +1,128 @@
+import os, sys
```

Comment:
I think this file can be named `models.py`.  The current file name `misc.py` does not really provide helpful documentation.